### DISCUSSION TITLE
Is it possible to build Image Magick on a mac Arm64 as Universal with x86_64

### DISCUSSION THREAD

**Discussion by potterpg** [ImageMagick/ImageMagick#5472] [General] [answered]
Is it possible to build Image Magick on a mac Arm64 as Universal with x86_64...

on a new Mac, using an Arm64 processor, xcode tends to build a universal build which includes an x86_64 version within the build.

How do you achieve this for Image Magick, or is it possible?

**Comment by Tekl** [comment]
Have you found a solution?

**Comment by dlemstra** [comment]
Can you clarify the question? I don't understand what is being asked.

**Reply by fmw42** [reply]
See https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary

**Reply by Tekl** [reply]
I was looking for a ./Configure command which builds a standalone and portable universal version of ImageMagick.

**Comment by potterpg** [comment, marked as answer]
Hi Tekl,
Kind of...
Compiling the static libraries as either arm64 or x86_64 and then manually joining the 2 using the lipo command
ie
`lipo ./arm64/libMagick++-7.Q16HDRI.a ./x86_64/libMagick++-7.Q16HDRI.a -create -output libMagick++-7.Q16HDRI.a`
This then gives you a universal build version of the library... which seems to work well.

**Comment by emendelson** [comment]
It's a bit late, but there's an answer here, complete with a download link:

https://stackoverflow.com/questions/77767021/how-to-build-a-portable-version-of-imagemagick-for-macos-monterey-ventura-sonoma/77767022#77767022
